test/integration/targets/inventory_aws_conformance/runme.sh

```diff
@@ -0,0 +1,152 @@
+  ec2_previous_state_code: backwardscompat | default(0)  # This is only returned with boto3 when starting, stopping, and terminating instances
+  ec2__in_monitoring_element: backwardscompat | default(false)  # double underscore not a typo. Trying to figure out wth this was.
+  ec2_requester_id: backwardscompat | default("") # FIXME not an attribute returned anymore on instances in boto3, will need code change
+  ec2_account_id: backwardscompat | default("") # FIXME not an attribute returned anymore on instances in boto3, will need code change
+  ec2_eventsSet: backwardscompat | default("")  # FIXME would need to add a describe_instance_status API call for instances to be able to add an events hostvar
+  ec2_persistent: backwardscompat | default(false)  # FIXME would need to add a describe_spot_instance_requests API call for any instances that have the SpotInstanceRequestId variable
+
+  # vars that change
+  ec2_block_devices: dict(block_device_mappings | map(attribute='device_name') | list | zip(block_device_mappings | map(attribute='ebs.volume_id') | list))
+  ec2_dns_name: public_dns_name
+  ec2_group_name: placement.group_name
+  ec2_id: instance_id
+  ec2_instance_profile: iam_instance_profile | default("")
+  ec2_ip_address: public_ip_address
+  ec2_kernel: kernel_id | default("")
+  ec2_monitored:  monitoring.state in ['enabled', 'pending']
+  ec2_monitoring_state: monitoring.state
+  ec2_owner_id: account_id
+  ec2_placement: placement.availability_zone
+  ec2_ramdisk: ramdisk_id | default("")
+  ec2_reason: state_transition_reason
+  ec2_security_group_ids: security_groups | map(attribute='group_id') | list |  join(',')
+  ec2_security_group_names: security_groups | map(attribute='name') | list |  join(',')
+  ec2_state: state.name
+  ec2_state_code: state.code
+  ec2_state_reason: state_reason.message if state_reason is defined else ""
+  ec2_sourceDestCheck: source_dest_check | lower | string  # butchered snake_case case not a typo.
+
+  # vars that just need ec2_ prefix
+  ec2_ami_launch_index: ami_launch_index | string
+  ec2_architecture: architecture
+  ec2_client_token: client_token
+  ec2_ebs_optimized: ebs_optimized
+  ec2_hypervisor: hypervisor
+  ec2_image_id: image_id
+  ec2_instance_type: instance_type
+  ec2_key_name: key_name
+  ec2_launch_time: 'launch_time | regex_replace(" ", "T") | regex_replace("(\+)(\d\d):(\d)(\d)$", ".\g<2>\g<3>Z")'
+  ec2_platform: platform | default("")
+  ec2_private_dns_name: private_dns_name
+  ec2_private_ip_address: private_ip_address
+  ec2_public_dns_name: public_dns_name
+  ec2_region: placement.region
+  ec2_root_device_name: root_device_name
+  ec2_root_device_type: root_device_type
+  ec2_spot_instance_request_id: spot_instance_request_id
+  ec2_subnet_id: subnet_id
+  ec2_virtualization_type: virtualization_type
+  ec2_vpc_id: vpc_id
+  tags: dict(tags.keys() | map('regex_replace', '[^A-Za-z0-9\_]', '_') | list | zip(tags.values() | list))
```

Comment:
I had put in `ec2_tag_Name: tags.Name`. I don't know 100% for sure if this is correct. It matches in my testing, but that doesn't guarantee that it will always match. I didn't think that `ec2.py` returned `tags`.

Reply:
It doesn't return tags. This is actually being used to compare the host vars; ec2.py script returned safe keys and unsafe values. The corner cases like this are pretty interesting.

Reply:
e.g. "ec2_tag_key_with_hyphens": "value:with:colons"
In this case hostvars are also influenced by replace_dash_in_groups = True.

Reply:
thanks, I can read that better now. So this takes the `tags` hostvar and scrubs non-ascii/underscore characters from the keys. Then it saves it back to the same `tags` key, and that's used elsewhere.